### PR TITLE
PR template addition

### PR DESCRIPTION
**Pull request by 0xc0170**
We would like to show users how a good pull request look like.  I used code highlight for the template, is that fine or better solutions ?

The main points that should be coevred:
- describe your changes (what is being changed and how)
- how it was tested (targets, toolchains, etc)
- pull request type (can be one of, not a mixture!)

The template is being updated here https://github.com/ARMmbed/mbed-os/pull/6078

### CODE REVIEW

File: docs/reference/contributing/guidelines/workflow.md [hunk 1]
```diff
@@ -51,6 +51,30 @@ Pull requests on GitHub have to meet the following requirements to keep the code
...
+A good example how a pull request template should look like.
+
+```
+Fix deep sleep locking bug
+
+# Description
+
+Fix problems which could leave deep sleep locked unintentionally, along with adding tests to verify this behavior is fixed.
+
+Tested locally with two targets, all toolchains.
```
Comment (cmonr) [inline docs/reference/contributing/guidelines/workflow.md]: I would recommend we be specific here, to encourage others to be specific as well.

Replace "two targets" with something like "K64F and K66F".

Comment (cmonr) [inline docs/reference/contributing/guidelines/workflow.md]: I would recommend we be specific here, to encourage others to be specific as well.

Replace "two targets" with something like "K64F and K66F".

File: docs/reference/contributing/guidelines/workflow.md [hunk 2]
```diff
@@ -51,6 +51,30 @@ Pull requests on GitHub have to meet the following requirements to keep the code
...
+
+Test results can be seen [here](just an example). 
+
+# Pull request type
+
+- [x] Fix
```
Comment (cmonr) [inline docs/reference/contributing/guidelines/workflow.md]: Like the checklist.

**Comment by 0xc0170** [comment]
@adbridge @theotherjimmy @kjbracey-arm Please review

**Comment by 0xc0170** [comment]
> Agree with Cruz about being specific as to which two targets were tested against. But otherwise looks like a nice simple example. Can always be refined down line, good to get something in.

I had it first (named two targets I randomly chose) , but did not want to pick up only two targets (which ones we would select), lets see how this goes and can add some info about being more specific

**Comment by AnotherButler** [comment]
Is it OK to merge this before 6078 merges?

**Comment by cmonr** [comment]
@AnotherButler Yup! It even seems like that PR is waiting on this! https://github.com/ARMmbed/mbed-os/pull/6078#issuecomment-365852699

**Pull request merged**